posts/inside-rust/2020-08-12-changes-to-x-py-defaults.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "Changes to x.py defaults"
+author: Joshua Nelson
+team: the compiler team <https://www.rust-lang.org/governance/teams/compiler>
+---
+
+Recently, the defaults for [`x.py`], the tool used to [bootstrap] the Rust compiler from source, changed. If you regularly contribute to Rust, this might affect your workflow.
+
+## What changes were made?
+
+- The default stage is now dependent on the subcommand:
+  + `dist`: stage 2
+  + `build`: stage 1
+  + `test`: stage 1
+  + `doc`: stage 0
+
+- stage 1 `rustc` artifacts are no longer built by `x.py build --stage 1`. To get the old behavior back, use `x.py build --stage 1 src/rustc`. The new behavior for `build --stage 1` builds everything except `rustc`, which includes the standard library, `rustdoc`, and various other tools (if the tools are enabled).
+
+- `debuginfo` now defaults to `1` when `debug = true`. Previously, the default was 2.
+
+## Why were the changes made?
+
+Previously, `x.py build` would build `rustc` three separate times:
+
+1. `build/stage0-rustc`
+2. `build/stage1-rustc`
+3. `build/stage2-rustc`
```

Comment:
This seems false - we only built 2 times, and uplifted stage 1 to stage 2 without full bootstrap being on, IIRC.

Reply:
Ah hmm I'm confused then - maybe libstd was being built three times? `--stage 2` means that _something_ was built 3 times, because libstd is built during all of stage 0, 1, and 2.

Reply:
As far as I know, the stage 2 libstd "build" is just a copy of the one built for stage 1.

Reply:
Well my changes seem much less impressive when you put it that way 😆 Updated to only say stage1 rustc/libstd